node/src/reactor.rs

```diff
@@ -443,9 +464,42 @@ where
         self.event_count += 1;
     }
 
+    /// Get both the allocated and total memory from sys-info + jemalloc
+    fn get_allocated_memory() -> Option<(u64, u64)> {
+        let mem_info = match sys_info::mem_info() {
+            Ok(mem_info) => mem_info,
+            Err(error) => {
+                tracing::error!(%error, "unable to get mem_info using sys-info");
+                return None;
+            }
+        };
+
+        let total = mem_info.total;
+
+        match epoch::mib() {
+            Ok(mib) => {
+                mib.advance().unwrap();
+            }
+            Err(error) => {
+                tracing::error!(%error, "unable to get epoch::mib from jemalloc");
+                return None;
+            }
+        }
```

Comment:
Can you add a comment why we have to do that?

Reply:
I found the answer in the documentation of `jemalloc-ctl`:
> many statistics are cached and only updated when the epoch is advanced.

Reply:
This seems to be an artifact of the internal implementation in `jemalloc_ctl` [here](https://docs.rs/jemalloc-ctl/0.3.3/jemalloc_ctl/stats/struct.allocated.html#method.mib) - I really just wanted to play nice and prevent a crash from `unwrap()`.

Reply:
Oh I understand, you mean the `epoch::advance()` call. Sorry

Reply:
I meant the `mib.advance()` part. It wasn't clear to me why we have to do that before reading the allocated memory. The documentation had an answer, can you add it? I  mean the part about caching statistics.